File: problems/arrays_and_hashing/problem_242/cpp/solution.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

class Solution
{
public:
    bool isAnagram(string s, string t)
    {
        if (s.length() != t.length())
            return false;
        vector<int> vec(26, 0);
        for (int i = 0; i < s.length(); ++i)
        {
            vec[s[i] - 'a']++;
            vec[t[i] - 'a']--;
        }
        for (int i = 0; i < 26; i++)
        {
            if (vec[i] != 0)
                return false;
        }
        return true;
    }
};

File: problems/arrays_and_hashing/problem_242/rust/solution.rs
struct Solution;

impl Solution {
    pub fn is_anagram(s: String, t: String) -> bool {
        if s.char().len() != t.char().len() {
            return false;
        }
        let mut vec: [0;26] = [];
        for elem in s {
            vec[elem - 'a'] += 1;
        }

        for elem in t {
            vec[elem - 'a'] -= 1;
        }

        for elem in vec {
            if elem != 0 {
                return false;
            }
        }

        true
    }
}
